opticdev/optic
# Connect up optic diff to new results format

## 🍗 Description
_What does this PR do? Anything folks should know?_

Ok so this is a beast of a PR. This all kind of needed to be done in one go, but this switches over `diff` and `diff_all` to use the new results format.

To hopefully make this easier to review, I've broken this up into a couple of parts to review separately:
- Updating the jsonChangelog function to work with the new results format https://github.com/opticdev/optic/pull/1501
- Updating the terminalChangelog function to work with the new results format https://github.com/opticdev/optic/pull/1504
- Updating the logComparison function to work with the new result format https://github.com/opticdev/optic/pull/1505
- Connecting up optic diff + diff all to all these changes https://github.com/opticdev/optic/pull/1506/files/c38935e96db6ad7153983020ea5173dd1a3b2285..6c389940d7fa2141ecf16f476fb3fe25a06f13a9


I'll want to publish a prerelease version here to play around and make sure we're not breaking anything major. The main things to note are:
- The JSON changelog format will have changed, but the information should be fairly similar
- There'll be some slight differences with the terminal changelog output, but should be similar outputs

TODO
- [x] Fix the tests
- [x] fix toUpperCase in custom rules issue
- [x] publish a prerelease


## 📚 References
_Links to relevant docs (Notion, Twist, GH issues, etc.), if applicable._

## 👹 QA
_How can other humans verify that this PR is correct?_

- Check the snapshots and see the same information is still surfaced
- try with prerelease - there's a prerelease version `0.36.6-1` that you can use 

things to test:
- custom rules
- spectral rules
- terminal output
- json point
- regular built in rules
